Make Weather Page more useful for Network Monitoring [1/2]

raleighlittles opened the pull request:
@markBowman asked that I submit a pull request for Task 2 (Scale pressure from min to max rather than from zero) on this story: https://www.pivotaltracker.com/n/projects/1314272/stories/156576199

Here's a screenshot of what the new chart looks like:

![screenshot from 2018-05-10 15-14-55](https://user-images.githubusercontent.com/2137201/39896914-fe3a9fac-5464-11e8-917d-5822ac301a94.png)

And for referece, here is what an old chart looked like on https://weather.lco.global/

![screenshot from 2018-05-10 15-16-39](https://user-images.githubusercontent.com/2137201/39896964-2bcc0172-5465-11e8-8758-083e22fe2b81.png)

Fingel commented on client/src/components/Site.vue:
```diff
@@ -183,9 +183,11 @@ export default {
+          limit:
+            {
+              // leave line below commented so that the pressure plot y axes auto-scales
+              //default: null
```
We shouldn't leave commented code in the source. Just remove the default: line and leave a comment explaining why it needs to be an empty object.

Fingel commented on client/src/components/Timeline.vue:
```diff
@@ -69,6 +69,10 @@ export default {
         },
         scales: {
+          yAxes: [{
+            beginAtZero: false
```
Won't this change the behaviour of all the charts that use Timeline.vue? Is that what we want?

hgielar replied:
@Fingel This was addressed in the second commit -- I added exceptions for certain units.

hgielar commented:
@markBowman @Fingel  In the latest commit I set up a small area above the 'Current Information' section to show the reasoning behind why we received a 'Weather Failure Reason' log, but some of the valueStrings are set as "None" for a few sites.

How come the telemetry is reporting a weather failure without specifying the reason? For example, from this url: https://weather-api.lco.global/query?site=ogg&datumname=Weather%20Failure%20Reason&start=2018-05-05T21:14:50Z&end=2018-05-11T00:14:50Z&agg=False

We get back:

```
[{"TimeStamp":"2018-05-05T22:02:41.152Z","Value":0,"ValueString":"Unknown"},{"TimeStamp":"2018-05-05T22:03:11.182Z","Value":0,"ValueString":"Sun Up"},{"TimeStamp":"2018-05-06T04:25:05.082Z","Value":0,"ValueString":"None"},{"TimeStamp":"2018-05-06T16:18:05.621Z","Value":0,"ValueString":"Sun Up"},{"TimeStamp":"2018-05-06T22:02:59.617Z","Value":0,"ValueString":"Unknown"},{"TimeStamp":"2018-05-06T22:03:29.647Z","Value":0,"ValueString":"Sun Up"},{"TimeStamp":"2018-05-07T04:25:38.294Z","Value":0,"ValueString":"Humidity alert"},{"TimeStamp":"2018-05-07T07:21:42.078Z","Value":0,"ValueString":"Dew point alert"},{"TimeStamp":"2018-05-07T07:23:27.116Z","Value":0,"ValueString":"The WMS is in SUSPEND_STATE"},{"TimeStamp":"2018-05-07T07:24:12.131Z","Value":0,"ValueString":"Dew point alert"},{"TimeStamp":"2018-05-07T07:24:42.143Z","Value":0,"ValueString":"The WMS is in SUSPEND_STATE"},{"TimeStamp":"2018-05-07T07:25:57.171Z","Value":0,"ValueString":"Dew point alert"},{"TimeStamp":"2018-05-07T07:26:12.176Z","Value":0,"ValueString":"The WMS is in SUSPEND_STATE"},{"TimeStamp":"2018-05-07T07:29:27.249Z","Value":0,"ValueString":"Dew point alert"},{"TimeStamp":"2018-05-07T07:30:42.276Z","Value":0,"ValueString":"The WMS is in SUSPEND_STATE"},{"TimeStamp":"2018-05-07T08:01:42.937Z","Value":0,"ValueString":"None"},{"TimeStamp":"2018-05-07T16:17:38.497Z","Value":0,"ValueString":"Sun Up"},{"TimeStamp":"2018-05-07T22:03:04.013Z","Value":0,"ValueString":"Unknown"},{"TimeStamp":"2018-05-07T22:03:34.040Z","Value":0,"ValueString":"Sun Up"},{"TimeStamp":"2018-05-08T04:25:57.721Z","Value":0,"ValueString":"Sky transparency too low"},{"TimeStamp":"2018-05-08T08:54:18.538Z","Value":0,"ValueString":"None"},{"TimeStamp":"2018-05-08T09:04:48.763Z","Value":0,"ValueString":"Sky transparency too low"},{"TimeStamp":"2018-05-08T09:42:34.554Z","Value":0,"ValueString":"None"},{"TimeStamp":"2018-05-08T13:49:24.788Z","Value":0,"ValueString":"Sky transparency too low"},{"TimeStamp":"2018-05-08T15:12:56.595Z","Value":0,"ValueString":"None"},{"TimeStamp":"2018-05-08T16:03:42.701Z","Value":0,"ValueString":"Sky transparency too low"},{"TimeStamp":"2018-05-08T16:17:13.001Z","Value":0,"ValueString":"Sun Up"},{"TimeStamp":"2018-05-08T22:02:56.278Z","Value":0,"ValueString":"Unknown"},{"TimeStamp":"2018-05-08T22:03:26.312Z","Value":0,"ValueString":"Sun Up"},{"TimeStamp":"2018-05-09T04:26:20.111Z","Value":0,"ValueString":"Sky transparency too low"},{"TimeStamp":"2018-05-09T05:02:35.925Z","Value":0,"ValueString":"Rain alert set"},{"TimeStamp":"2018-05-09T05:04:50.986Z","Value":0,"ValueString":"The WMS is in SUSPEND_STATE"},{"TimeStamp":"2018-05-09T05:05:20.997Z","Value":0,"ValueString":"Rain alert set"},{"TimeStamp":"2018-05-09T05:08:06.062Z","Value":0,"ValueString":"The WMS is in SUSPEND_STATE"},{"TimeStamp":"2018-05-09T05:09:06.083Z","Value":0,"ValueString":"Sky transparency too low"},{"TimeStamp":"2018-05-09T05:13:21.182Z","Value":0,"ValueString":"Rain alert set"},{"TimeStamp":"2018-05-09T05:15:36.230Z","Value":0,"ValueString":"The WMS is in SUSPEND_STATE"},{"TimeStamp":"2018-05-09T05:16:36.250Z","Value":0,"ValueString":"Sky transparency too low"},{"TimeStamp":"2018-05-09T06:46:23.241Z","Value":0,"ValueString":"Rain alert set"},{"TimeStamp":"2018-05-09T06:48:38.288Z","Value":0,"ValueString":"The WMS is in SUSPEND_STATE"},{"TimeStamp":"2018-05-09T06:49:38.310Z","Value":0,"ValueString":"Sky transparency too low"},{"TimeStamp":"2018-05-09T10:15:57.759Z","Value":0,"ValueString":"None"},{"TimeStamp":"2018-05-09T16:16:50.742Z","Value":0,"ValueString":"Sun Up"},{"TimeStamp":"2018-05-09T22:02:50.738Z","Value":0,"ValueString":"Unknown"},{"TimeStamp":"2018-05-09T22:03:20.772Z","Value":0,"ValueString":"Sun Up"},{"TimeStamp":"2018-05-10T04:26:30.121Z","Value":0,"ValueString":"None"},{"TimeStamp":"2018-05-10T16:16:16.607Z","Value":0,"ValueString":"Sun Up"},{"TimeStamp":"2018-05-10T22:03:15.129Z","Value":0,"ValueString":"Unknown"},{"TimeStamp":"2018-05-10T22:03:45.157Z","Value":0,"ValueString":"Sun Up"}]
```

hgielar commented:
Some screenshots of what the new UI changes are. 
![screenshot from 2018-05-10 18-13-10](https://user-images.githubusercontent.com/2137201/39901520-e686e71c-547d-11e8-9a1b-60eab7a8bd87.png)
![screenshot from 2018-05-10 18-14-12](https://user-images.githubusercontent.com/2137201/39901530-fcf6613a-547d-11e8-8743-c6fd77d0c5e5.png)
![screenshot from 2018-05-10 18-14-53](https://user-images.githubusercontent.com/2137201/39901547-14e75952-547e-11e8-93de-f4bf30eb279b.png)

wrosing commented:
I am a bit confused.  It appears we are promoting Sky Transparency over Sky
temperature.  The latter is a objective measure of an input from a
detector.  Unless someone has invented something different, our
transparency measurement is a bit of numerical wizardry that has not been
systematically calibrated.  Raleigh I know you are new so Mark B. may be
able to clarify.

I am not sure why the dewpoint it is being 'pivoted' around zero C.  Useful
plots of dewpoint should IMHO be over-plotted with the
outside temperature, a key optic temperature, the dewpoint and humidity.
 Then you can see clearly if humidity is rising and the dewpoint
is racing towards a key optic temperature that is is time to do something
-- usually close in our case.

On Fri, May 11, 2018 at 1:15 AM, Raleigh Littles <notifications@github.com>
wrote:

> Some screenshots of what the new UI changes are.
> [image: screenshot from 2018-05-10 18-13-10]
> <https://user-images.githubusercontent.com/2137201/39901520-e686e71c-547d-11e8-9a1b-60eab7a8bd87.png>
> [image: screenshot from 2018-05-10 18-14-12]
> <https://user-images.githubusercontent.com/2137201/39901530-fcf6613a-547d-11e8-8743-c6fd77d0c5e5.png>
> [image: screenshot from 2018-05-10 18-14-53]
> <https://user-images.githubusercontent.com/2137201/39901547-14e75952-547e-11e8-93de-f4bf30eb279b.png>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/LCOGT/weather/pull/5#issuecomment-388230882>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKedsIKy1cQRH8kqQLSO2U216BZXmHoRks5txOYqgaJpZM4T6opb>
> .
>



-- 
Clear skies,

Wayne Rosing
Founder
805 708 6901 cell
805 880 1603 office

Please note wrosing@lco.global is our preferred address.

Fingel commented on client/src/components/Site.vue:
```diff
@@ -86,13 +102,13 @@
     <section class="section section-xsmall ">
-      <h4 class="is-size-4">Sky - Ambient Temp
-        <a class="helptoggle is-pulled-right"><sup><small>?</small></sup></a><span class="help is-pulled-right">Sky Temperature is inferred from 8-14µm irradiance measure by a Boltwood II cloud sensor at the site's weather station.</span>
+      <h4 class="is-size-4">Sky Transparency*
```
What is the asterisk for?

raleighlittles replied:
Basically just to show that the sky transparency is a calculated value and it isn't measured directly. (see task 4: *Switch sky-ambient plot to be transparency but make it clear it is a calculated value not measured directly, or plot both sky-ambient and transparency on same graph. Add threshold line in red.*) 

I wasn't quite sure what would be the best way to distinguish the sky transparency thing as being a calculated value. Should I just mention that fact in the dropdown instead? Or color the dropdown icon differently or something? The asterisk was just the first thing that came to mind.

Fingel commented on client/src/components/Site.vue:
```diff
@@ -16,11 +16,21 @@
       </p>
     </div>
     <div>
+      <p class="level-heading heading"> Latest Closure Information </p>
```
This is good, but it wastes a lot of space on the screen (an entire graph's worth). Can we try moving this somewhere? Perhaps on the same line as the "OK to open" title. Or somewhere near/on that graph.

Fingel commented:
I don't know why the site telemetry contains "None" instead of "Unknown" for the closure reason sometimes, but I my initial thought would be to treat them as the same. Perhaps translate "None" to "Unknown" in the UI. Or ask @markBowman or @sfoale if there is an actual difference between the values.

Fingel commented:
Raleigh, I just had a converstaion with Todd about some of this stuff. Before you do any more work, let's talk. He had some good ideas.

markBowman commented:
@wrosing I agree that Sky Transparency is not a great measure but it is the measure LCO uses to make close/open decisions. As long as this is the case we should show transparency and the closure threshold on the Weather Page so that we can see why the enclosures are open or closed (as requested by Todd).

markBowman commented:
@raleighlittles I believe the "Weather Failure Reason" is set to "Unknown" when the software starts (it has this value at the same time every day). "None" indicates that the weather is good. So, for reporting, "None" and "Unknown" should be treated as "Weather is good"

raleighlittles commented:
I'm having quite a bit of trouble setting up the sunset/sunrise graph times for the Ok To Open Plot (Task 7/8) in 24 hour mode. I am able to get the correct times (this is easy since its already done a few other places in the codebase), but I can't figure out why the time being reported isn't displayed properly on the graph.

For example, here's what I see in the log, for Santa Barbara:


webpack-internal:///160:70 sunrise time in UTC
webpack-internal:///160:73 2018/05/13 12:59:53
webpack-internal:///160:75 sunrise time with the offset of: -8
webpack-internal:///160:77 2018/05/13 04:59:53
webpack-internal:///160:79 sunset time in UTC, with yesterdays day
webpack-internal:///160:80 2018/05/13 02:54:25
webpack-internal:///160:82 sunset time with the offset of: -8 (and day -1)
webpack-internal:///160:83 2018/05/12 18:54:25


The UTC times correctly match up with what is displayed on the site page (http://localhost:8080/#/bpl), but when you look at the graph, these times obviously aren't being ploted -- instead, there's a line at 10:00 UTC for sunset and a line at 20:00 UTC for sunrise, which, if you'll notice, is about 8 hours off from the correct value. I assumed that this meant that the graph was dispalying the data in local time for some reason, so I figured I'd apply a manual UTC offset (based on whatever site it was) to correct this issue. However, this doesn't work -- the offset being applied does bring the BPL times closer to the right value, but it still isn't correct for all sites, even though the offset is being computed based on whatever timezone each site is in (meaning that there isn't an issue with the timezones). I'm pretty much at a loss as to why this isn't working, the time being displayed in the console log is not the correct time shown on the graph even though I literally copy and pasted them. At this point I'm guessing there's something really obvious that I'm missing and its kind of driving me crazy.

Here's the two screenshots, the first is with the offset disabled (this is what should work, just native UTC time), and then the offset enabled. Neither of them have the correct times on either the sunset or the sunrise time. 

![bpl ok to open_utc](https://user-images.githubusercontent.com/2137201/39977142-ad83f07e-56ec-11e8-9c9c-0fe3986032d3.png)
![bpl screenshot utc offset on](https://user-images.githubusercontent.com/2137201/39977145-b1a3ab4a-56ec-11e8-825c-2e5811d269a4.png)


Any help is appreciated @markBowman @Fingel.

Fingel commented:
Hey Raleigh,
I'd be happy to pair with you on this next time you're in.

Fingel commented on client/src/components/WeatherNav.vue:
```diff
@@ -65,7 +67,15 @@
+    },
+    methods:
+      {
+        onHandleStatus(status)
```
What is this used for?

Fingel commented on client/src/timeline.js:
```diff
@@ -404,24 +409,28 @@ Chart.defaults.timeline = {
             categoryPercentage: 0.8,
             barPercentage: 0.9,
             offset: true,
+          //offset: false,
```
Can we just remove lines instead of commenting them out? Same for lines 414 and 415 and 432

Fingel commented on client/src/components/NavItem.vue:
```diff
@@ -19,7 +19,13 @@
+    props: ['site', 'sitestatus'],
+    data: function() {
+
+      return {site_code: this.site.code, site_status: '??'}
```
Is site_status used anywhere?

Fingel commented on client/src/components/Timeline.vue:
```diff
@@ -2,20 +2,24 @@
+
       const open = this.cdata.filter(item => item.ValueString === 'true').map(
-        point => moment.utc(point.TimeStamp, 'YYYY/MM/DD HH:mm:ss')
+        point => (moment.utc(point.TimeStamp, 'YYYY/MM/DD HH:mm:ss'))
```
extra parentheses? Line 20 too.

raleighlittles replied:
Have been removed.

Fingel commented on client/src/components/Timeline.vue:
```diff
@@ -2,20 +2,24 @@
 export default {
   name: 'Timeline',
-  props: ['cdata', 'datumid', 'datumname'],
+  props: ['cdata', 'datumid', 'datumname','suntimes', 'timezone'],
```
Where is the timezone used?

raleighlittles replied:
It was initially used when I was trying to implement the sunrise/sunset annotations, but it is no longer used -- it has been removed in the latest commit.

raleighlittles commented on client/src/components/Site.vue:
```diff
@@ -293,14 +358,33 @@ export default {
     sunrise(){
-      return moment.utc((this.suntTimes.sunrise)).format('HH:mm');
+      // get last sunrise
+      return moment.utc((this.suntTimes.slice(-1)[0].sunrise)).format('HH:mm');
```
Good point! I modified this, they're now formatted as String when used in the HTML template.

raleighlittles commented on client/src/components/Timeline.vue:
```diff
@@ -45,12 +73,57 @@ export default {
+          label: {
+            enabled: true,
+            position: "top",
+            content: (this.suntimes.length > 2) ? ("") : ("sunrise")
```
Not particularly, I just use them to aid readability for me. They were removed in the latest commit.

raleighlittles commented:
@markBowman Thanks for the feedback.

1. I don't know what the sky transparency threshold should be (didn't see this mentioned on the story), but I added a temporary threshold line at 75%.

2. I added the dewpoint closure thresholds, following your formula of sky temperature minus 2 degrees.

3. I moved the sky temperature plot to be above the sky transparency plot. When I am finished with the ticket overall, I'll try to create the multi-axis plot.

4. I doubled the size of the blue bars in the Ok to Open plot. 

-------

Screenshots showing some changes.
![screenshot from 2018-05-18 17-35-19](https://user-images.githubusercontent.com/2137201/40263065-1eda90d8-5ac2-11e8-93cd-11e6e95edddf.png)
![screenshot from 2018-05-18 17-36-16](https://user-images.githubusercontent.com/2137201/40263066-1ef1fef8-5ac2-11e8-81a0-95ff18f50ad8.png)
![screenshot from 2018-05-18 17-36-47](https://user-images.githubusercontent.com/2137201/40263067-1f0a668c-5ac2-11e8-9f84-825aecf1917a.png)

wrosing commented:
I wilL now do my broken record routine:  But these new plots help make the
point.

If you look at the math, sky transparency is t = a +b*(sky-ambient).  Fine,
some high number feels better than a negative number, but at then end of
the day we are scientists not children.

What is important is the thresholds chosen to derive from the number.
Based on a lot more experience than we have, Boltwoods parse the sky
into Clear, Cloudy, Very Cloudy and WET.  I think that is what we can
expect from them.  Opening in either of the last two cases is problematic.

I still believe Dew point is best represented as a differential from
outside ambient rather than 0.0  C -- and also represented with respect to
M1 temperature.  We simply do not care if DP is -2C below either optic
temperature.  The optics care a lot if DP is close to even or above their
temperature.

On Sat, May 19, 2018 at 12:37 AM, Raleigh Littles <notifications@github.com>
wrote:

> @markBowman <https://github.com/markBowman> Thanks for the feedback.
>
>    1.
>
>    I don't know what the sky transparency threshold should be (didn't see
>    this mentioned on the story), but I added a temporary threshold line at 75%.
>    2.
>
>    I added the dewpoint closure thresholds, following your formula of sky
>    temperature minus 2 degrees.
>    3.
>
>    I moved the sky temperature plot to be above the sky transparency
>    plot. When I am finished with the ticket overall, I'll try to create the
>    multi-axis plot.
>    4.
>
>    I doubled the size of the blue bars in the Ok to Open plot.
>
> ------------------------------
>
> Screenshots showing some changes.
> [image: screenshot from 2018-05-18 17-35-19]
> <https://user-images.githubusercontent.com/2137201/40263065-1eda90d8-5ac2-11e8-93cd-11e6e95edddf.png>
> [image: screenshot from 2018-05-18 17-36-16]
> <https://user-images.githubusercontent.com/2137201/40263066-1ef1fef8-5ac2-11e8-81a0-95ff18f50ad8.png>
> [image: screenshot from 2018-05-18 17-36-47]
> <https://user-images.githubusercontent.com/2137201/40263067-1f0a668c-5ac2-11e8-9f84-825aecf1917a.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LCOGT/weather/pull/5#issuecomment-390365147>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKedsJjuJLZBGKFQXC3KDZi7a_EGFEdWks5tz2lUgaJpZM4T6opb>
> .
>



-- 
Clear skies,

Wayne Rosing
Founder
805 708 6901 cell
805 880 1603 office

Please note wrosing@lco.global is our preferred address.

markBowman commented:
Raleigh,
* The threshold for sky transparency should come from the datum "Boltwood Transparency Close Threshold" which is available in ES for each site.
* We should change the dewpoint plot to be (Air Temperature - Dew Point) with a threshold value of 2 degrees

Wayne,
I know this isn't exactly what you want:
* at the moment Sky Transparency is the closure criteria so the Weather page should relate to the code/policy
* to plot dew-point versus M1 temperature we would have to either chose 1 telescope at each site or add a lot more graphs. The primary purpose of the weather page is to see if a site is open/closed and why, so I hope we can omit the M1-dewpoint graph.

wrosing commented:
I realize we are historically stuck with Transparency.

M1 temp compared to dewpoint should be the most accurate way to decide if
closing to protect M1 is the goal. Given we have multiple M1's I guess the
best scheme is pick the one closest to dewing and decide that way.  If we
are making the decision based on the Viasala probe and humidity reported
inside the dome, and a dewpoint derived from THAT data then so be it,
that is reasonably close physically.

As for displaying... real estate is valuable.

On Mon, May 21, 2018 at 9:54 PM, markBowman <notifications@github.com>
wrote:

> Raleigh,
>
>    - The threshold for sky transparency should come from the datum
>    "Boltwood Transparency Close Threshold" which is available in ES for each
>    site.
>    - We should change the dewpoint plot to be (Air Temperature - Dew
>    Point) with a threshold value of 2 degrees
>
> Wayne,
> I know this isn't exactly what you want:
>
>    - at the moment Sky Transparency is the closure criteria so the
>    Weather page should relate to the code/policy
>    - to plot dew-point versus M1 temperature we would have to either
>    chose 1 telescope at each site or add a lot more graphs. The primary
>    purpose of the weather page is to see if a site is open/closed and why, so
>    I hope we can omit the M1-dewpoint graph.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LCOGT/weather/pull/5#issuecomment-390794622>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKedsHcgGC4onnrdTmlYw2ZAl-o5PTbiks5t0zd-gaJpZM4T6opb>
> .
>



-- 
Clear skies,

Wayne Rosing
Founder
805 708 6901 cell
805 880 1603 office

Please note wrosing@lco.global is our preferred address.
